### DISCUSSION TITLE
Action Plan Automation, Flow via CreateActionPlanfromTemplate

### DISCUSSION THREAD

**Discussion by tatertotsey** [SalesforceLabs/ActionPlansV4#109] [Q&A]
Hi, 
I am trying to create an Action Plan depending on pre-created Action Plan Template via flow. 

If an Opportunity's stage changed into Closed, the flow is triggered and creates Action Plan from Template. The issue is, when the Action Plan is created via flow, the controlling fields of the tasks become different from the chosen template. 

If I create the Action Plan manually, the depending fields seems exact to the ones in the Template, so no problem there. 

What should be the solution to this, what am I missing while creating the flow? 
Thank you!

**Comment by dschach** [comment]
Thanks for raising this. Because the original (Visualforce) code relies on lots of Visualforce Apex, we had to do a rewrite to make the invocable Apex. If I'm hearing you correctly, the controlling tasks are wrong. Is that true?

Looks like we need to release a fix if that's the case.

**Reply by tatertotsey** [reply]
Exactly, the controlling tasks are wrong.

**Comment by filkanczula** [comment]
Hi, I am using a Dev Sandbox in a Professional Edition, I am a System Admin, but I cannot see Apex Actions for Action Plans in the flow builder. What could be the reason? I have granted myself Action Plan Admin PS to be sure it's not it, but it hasn't helped.
